### PR TITLE
Fix the text editor theme not being applied on editor start

### LINKED ISSUE #88713: Script Editor errors are white
### Tested versions

4.3 87d40ba
Does not happen in 4.3 dev3.

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

image

### Steps to reproduce

1. Create script
2. Make error

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by MajorMcDoom**
Fixes #88713 

This was likely a regression caused by #88474 .

### CODE REVIEW

File: editor/code_editor.cpp [hunk 1]
```diff
@@ -1683,10 +1683,6 @@ void CodeTextEditor::goto_error() {
 }
 
 void CodeTextEditor::_update_text_editor_theme() {
-	if (!EditorThemeManager::is_generated_theme_outdated()) {
```
Comment (Mickeon) [inline editor/code_editor.cpp:1686]: Won't the removal of this check cause a whole slew of unnecessary updating?

**Review by Mickeon — COMMENTED** [review]
Oopsie.

File: editor/code_editor.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/code_editor.cpp:1686]: This seems to be called only from `NOTIFICATION_THEME_CHANGED`, so the condition was wrongly added .-. (I blame the rebase)

File: editor/code_editor.cpp [hunk 2]
```diff
@@ -1059,7 +1059,7 @@ void CodeTextEditor::update_editor_settings() {
 		text_editor->set_line_length_guidelines(TypedArray<int>());
 	}
 
-	_update_font_ligatures();
+	_update_text_editor_theme();
```
Comment (KoBeWi) [inline editor/code_editor.cpp:1062]: Pretty sure this is unnecessary. Or at least it should be inside the condition you removed below.

File: editor/code_editor.cpp (hunk 1 above)
Comment (Mickeon) [reply editor/code_editor.cpp:1686]: Thus is what happens with last-minute rebases. A beautiful experience.

**Comment by MajorMcDoom** [comment]
Hate to admit it but I'm a little out of my depths here.  That rebase definitely left me feeling very unsure. Could someone with more experience do a proper fix for this bug? I don't want to accidentally introduce other issues.

**Comment by KoBeWi** [comment]
By rebase I meant my PR #69032, which might've actually introduced the issue. The `is_generated_theme_outdated()` is used in a code path that doesn't involve editor settings, so I think it changed when I rebased my PR and I missed that it's no longer needed here. Deleting the condition is probably sufficient to fix the issue (I didn't test though).

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**